From 9f10400ea754bc5516b56d02f16cf4ae72151a9e Mon Sep 17 00:00:00 2001
From: Geetha sowjanya <gakula@marvell.com>
Date: Wed, 10 Apr 2019 11:38:44 +0530
Subject: [PATCH 143/386] octeontx2-af: Fix enabling/disabling pause frames
 generation

Due to bug in present code enable/disable pause frames
generation works only for lmac0. This patch fixes the issue.

Change-Id: Icf0124afd33aca3a5b6537372c74c34b9a897dc3
Signed-off-by: Geetha sowjanya <gakula@marvell.com>
Reported-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/7167
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/cgx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
index 98dd6b1dc575..b78e7a1302e1 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
@@ -448,14 +448,14 @@ int cgx_lmac_set_pause_frm(void *cgxd, int lmac_id,
 	cfg |= tx_pause ? CGX_SMUX_TX_CTL_L2P_BP_CONV : 0x0;
 	cgx_write(cgx, lmac_id, CGXX_SMUX_TX_CTL, cfg);
 
-	cfg = cgx_read(cgx, lmac_id, CGXX_CMR_RX_OVR_BP);
+	cfg = cgx_read(cgx, 0, CGXX_CMR_RX_OVR_BP);
 	if (tx_pause) {
 		cfg &= ~CGX_CMR_RX_OVR_BP_EN(lmac_id);
 	} else {
 		cfg |= CGX_CMR_RX_OVR_BP_EN(lmac_id);
 		cfg &= ~CGX_CMR_RX_OVR_BP_BP(lmac_id);
 	}
-	cgx_write(cgx, lmac_id, CGXX_CMR_RX_OVR_BP, cfg);
+	cgx_write(cgx, 0, CGXX_CMR_RX_OVR_BP, cfg);
 	return 0;
 }
 EXPORT_SYMBOL(cgx_lmac_set_pause_frm);
-- 
2.17.1

